pom.xml

```diff
@@ -113,6 +113,7 @@
     <spark.bundle.hive.shade.prefix></spark.bundle.hive.shade.prefix>
     <utilities.bundle.hive.scope>provided</utilities.bundle.hive.scope>
     <utilities.bundle.hive.shade.prefix></utilities.bundle.hive.shade.prefix>
+    <argLine>-Xmx1024m -XX:MaxPermSize=256m</argLine>
```

Comment:
@prashantwason this has a side effect of making the builds run slower. The reason is that when a build is submitted in travis, it needs to wait for resources to be available for the build to get triggered. My understanding is that if we ask for a larger JVM for non-IntegrationTest modules as well, this will make the process slower.
How much more memory is required for the coverage report ?

Reply:
I don't think travis has the smarts to grok this and get the resource needs?  We just get a VM with some fixed RAM https://docs.travis-ci.com/user/reference/overview/ 

`-Xmx` won't also grab the 1GB right off the bat.. do you see the builds slowing down? would be interesting to understand why..

Reply:
I did not add this line. It was already part of the maven-surefire plugin's argLine.  I just moved the definition to a property so that jacoco plugin can add its own java-agent config to it and the unit tests can execute with jacococ agent enabled. 

This was added in the following commit 3 months ago:
https://github.com/apache/incubator-hudi/pull/979

Reply:
Okay, makes sense. 

@vinothchandar I don't see it slowing down, just had a hunch, merging this as this does not seem to have any side-effect.